src/Chainweb/Version.hs

```diff
@@ -205,20 +203,15 @@ instance IsMerkleLogEntry ChainwebHashTag ChainwebVersion where
     {-# INLINE toMerkleNode #-}
     {-# INLINE fromMerkleNode #-}
 
-chainwebVersionToText :: ChainwebVersion -> T.Text
-chainwebVersionToText v =
-    case HM.lookup v prettyVersions of
-        Just t -> t
-        Nothing -> case v of
-            Test{} -> "test"
-            TestWithTime{} -> "testWithTime"
-            TestWithPow{} -> "testWithPow"
-            Testnet00 -> "testnet00"
+chainwebVersionToText :: HasCallStack => ChainwebVersion -> T.Text
+chainwebVersionToText Testnet00 = "testnet00"
+chainwebVersionToText v = fromJuste $ HM.lookup v prettyVersions
```

Comment:
Hm, this will be a source of silent pattern match failures. When someone creates a new production `ChainwebVersion` value, they won't be warned about the missing match here.

Reply:
Hopefully quickcheck would catch it.